### DISCUSSION TITLE
Word naming task plugin

### DISCUSSION THREAD

**Discussion by 76Darya** [jspsych/jsPsych#2965] [Q&A]
Hi there، I have problem with my project task، I'm going to creat a word naming task with jspsych and jatos, but I dont know how to find its plugins and scripts

**Comment by ChristopheBossens** [comment]
What have you tried already? I experimented with this system in the past and from what I remember you need to have a local installation of Jatos which provides the jatos scripts. You add all the jspsych scripts yourself and when ready you export everything and import on a remote server.

**Reply by 76Darya** [reply]
Actually I am going to collect words' reaction time data as well as recorded voice for word' accuracy... but I dont know how to do it. 
I tried one time but I couldnt add record plugin correctly، the data didnt save on the server too.

**Reply by ChristopheBossens** [reply]
To work with the microphone you need to include the jsPsychInitializeMicrophone plugin, did you do this?

What development environment are you using? Using Visual Studio Code with the Live Server option, I was able to launch a jsPsychHtmlAudioResponse trial which collects audio data. Just to test, I was able to upload that code to cognition.run, where I was also able to get the audio response

Could you share any code that you have already tried? If we can get it working on your local computer it should also work in a server environment.

**Reply by 76Darya** [reply]
I used Visual Studio,
 yes I can share it, would you please give me your email address?

**Comment by ChristopheBossens** [comment]
***@***.***

Op di 14 mrt. 2023 22:26 schreef 76Darya ***@***.***>:

**Reply by 76Darya** [reply]
unfortunately your email address isn't clear, I Tried to copy paste the code here but it doesn' paste

**Reply by 76Darya** [reply]
<!DOCTYPE html>
<html>
  <head>
    <script src="jspsych/jspsych.js"></script>
    <script src="jspsych/plugin-survey-text.js"></script>
    <script src="jspsych/plugin-preload.js"></script>
    <script src="jspsych/plugin-html-keyboard-response.js"></script>
    <script src="word1.js"></script>
    <script src="word2.js"></script>
    <script src="word3.js"></script>
    <script src="word4.js"></script>
    <link href="jspsych/jspsych.css" rel="stylesheet" type="text/css" />
  </head>
  <body></body>
  <script>

    var jsPsych = initJsPsych({
      case_sensitive_responses: true, // to ignore the letter case of a keyboard response, set this parameter to false or remove it
      on_finish: function() {
        jsPsych.data.displayData();
      }
    }); 
    var preload = {
      type: jsPsychPreload,
      auto_preload: true
    };
    var welcome = {
      type: jsPsychHtmlKeyboardResponse,
    choices: "ALL_KEYS",
      stimulus: '<p> Hello,  Thanks for your participation, press any key you want.
     
    };

    var instructions = {
      type: jsPsychHtmlKeyboardResponse,
    choices: "ALL_KEYS",
      stimulus: '<p> This test consist 108 words, you have to read them aloud and press the space key as quickly as you can, each word would be on screen for 2 seconds. You have 10 second to record your voice. To continue, press any key you want.
      
    };
  var hint0 = {
      type: jsPsychHtmlKeyboardResponse,
      choices: " ",
      stimulus: '<h1>(SPACE) press a key to start </h1>',
    };
  var hint1 = {
      type: jsPsychHtmlKeyboardResponse,
      choices: "ALL_KEYS",
      stimulus: '<h1>time for a break</h1>',
    };
    var hint2 = {
      type: jsPsychHtmlKeyboardResponse,
      choices: "ALL_KEYS",
      stimulus: '<h1> time for a break</h1>',
    };
    var hint3 = {
      type: jsPsychHtmlKeyboardResponse,
      choices: "ALL_KEYS",
      stimulus: '<h1> time for a break</h1>',
    };
    var hint4 = {
      type: jsPsychHtmlKeyboardResponse,
      choices: "ALL_KEYS",
      stimulus: '<h1> time for a break</h1>',
    };
    var hint5 = {
      type: jsPsychHtmlKeyboardResponse,
      choices: "ALL_KEYS",
      stimulus: '<h1> Finish    };
  var name_entry = {
      type: jsPsychSurveyText,
      questions: [
        {prompt: '', placeholder: ' ', columns: 50, name: 'name'}
      ],
    };

  var test_stimuli0 = [
  { stimulus: "teapot"},
  { stimulus: "pigeon"},
  { stimulus: "strife"},
  { stimulus: "sinful"}
   ];

**Reply by 76Darya** [reply]
That's a part of my script, but with out the recording plugin... also I dont know how to set a time for words... and at the end it cant save the data on the server

**Reply by ChristopheBossens** [reply]
What I do in these cases is first focus on the trial that is giving you trouble :). Just to be clear, from what I understood from your explanation so far you are going to present a word on the screen, and the participant needs to read that word as fast as possible?

The code below allows you to do that (never mind some of the parameters that you require in addition, I think for now it is important to check if you can get the audio response). Can you see if this works for you? I don't have experience with using Jatos, but if I use Visual Studio Code with the live server plugin I am able to see the audio response. I have also tested this code on cognition.run where I can also get the audio response.

```
  <script>
    const jsPsych = initJsPsych({
      case_sensitive_responses: true, // to ignore the letter case of a keyboard response, set this parameter to false or remove it
      on_finish: function() {
            jsPsych.data.displayData();
        }
    });

    var initialize_microphone_trial = {
    type: jsPsychInitializeMicrophone
};

    let start_trial ={
        type: jsPsychHtmlKeyboardResponse,
        stimulus: 'Press a key to start the audio recording trial',
    }

    let audio_trial = {
        type: jsPsychHtmlAudioResponse,
        stimulus: `
        <p style="font-size:48px; color:red;">GREEN</p>
        <p>Speak the color of the ink.</p>`,
        recording_duration: 3500
    }

    jsPsych.run([initialize_microphone_trial, start_trial, audio_trial]);
  </script>
```

**Comment by 76Darya** [comment]
Hi.. I tried to send you an Email but it was unsuccessful... Do you have
any accounts on skype? if it is possible, I can call you there... I
couldn't use the codes and there is no enough time for me

**Reply by ChristopheBossens** [reply]
Hi,

Can you try christophe@copsy.be, and send me the code that you currently have?

Just to be clear, I am not familiar with the JATOS platform so if you need to set it up there I cannot really help.
